src/main/java/com/microsoftopentechnologies/windowsazurestorage/service/UploadToBlobService.java

```diff
@@ -179,6 +179,8 @@ private CloudBlobContainer getCloudBlobContainer() throws URISyntaxException, St
         // Delete previous contents if cleanup is needed
         if (serviceData.isCleanUpContainerOrShare()) {
             deleteBlobs(container.listBlobs());
+        } else if (serviceData.isCleanUpVirtualPath() && StringUtils.isNotBlank(serviceData.getVirtualPath())) {
+            deleteBlobs(container.getDirectoryReference(serviceData.getVirtualPath()).listBlobs());
```

Comment:
`container.getDirectoryReference(serviceData.getVirtualPath())` may get exception or return when the `virtualPath` is invalid or not exists. I think it needs more check here.

Reply:
If `virtualPath` is valid but no blobs exist, the `Iterable` that's passed to `deleteBlobs` will be empty and thus `deleteBlobs` won't do anything - which is the correct behavior for a new `virtualPath`.

If `virtualPath` is invalid, an exception will be thrown when iterating (because the elements are loaded lazily). This exception will be caught by `uploadIndividuals` / `uploadArchive`.